### DISCUSSION TITLE
Missing file

### DISCUSSION THREAD

**Discussion by antoscha** [su2code/SU2#1547] [Q&A]
Hi,

A file is missing in 7.3.0, and compilation breaks:

```
In file included from ../src/../include/SU2_CFD.hpp:40,
                 from ../src/SU2_CFD.cpp:28:
../src/../include/output/COutput.hpp:50:10: fatal error: mel.hpp: No such file or directory
   50 | #include "mel.hpp"
      |          ^~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:11862: ../src/bin_SU2_CFD-SU2_CFD.o] Error 1
```

**Comment by pcarruscag** [comment]
Hi,
It looks like you are using the old build system. Meson downloads that dependency automatically.
Is there a particular reason why you are still using the old `preconfigure.py && make` process?

**Reply by antoscha** [reply]
Yes, probably. I consider `./configure && make` as a conventional and pretty convenient process. Is it broken in latest releases?

**Reply by pcarruscag** [reply]
It has been legacy for a long time. Please use the recommended (in the documentation) way of building SU2. The code will likely run faster that way.

**Reply by bigfooted** [reply]
Should we remove the option to configure/compile like this?

**Reply by pcarruscag** [reply]
I was thinking of moving the files to some folder, if someone wants to use them they can copy.
Then as folks start to complain we can figure out why they are not using meson, and fix it if it needs fixing.

**Reply by antoscha** [reply]
I tried using meson but the problem persists.

**Reply by antoscha** [reply]
New 7.3.1 has the same error, nothing changed. Also included `meson.py` can't even start compiling.

**Reply by pcarruscag** [reply]
Do you have git? Are you doing git clone? or downloading the code manually?
Is this on windows, linux? Is your firewall blocking the download of the additional dependencies?
Are you doing this on a machine without internet access?
Have you tried deleting the SU2 folder and starting from scratch?
Can you post the complete output of the process?

**Reply by antoscha** [reply]
I try to maintain [this](https://aur.archlinux.org/packages/su2). Until recent times it was not that hard. My machine has Internet access and of course able to connect to Github. The package is built in a clean chroot acquiring all necessary dependencies.

**Reply by pcarruscag** [reply]
@antoscha are you doing `meson ...` or `./meson.py ...` ? The former will not work you need to use the meson.py script that ships with SU2.

**Reply by carlosal1015** [reply]
I can confirm [the same error](https://asciinema.org/a/490297).

**Reply by pcarruscag** [reply]
It is not an error, you need to use the meson.py script in the root of the code, not the system version of meson.
So, `./meson.py`, not `meson` exactly what we say here https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/#quick-compilation-guide

**Reply by carlosal1015** [reply]
It is true, thank you very much for this precious work.

@antoscha if you want try since today, [the AUR package has been updated](https://aur.archlinux.org/cgit/aur.git/commit/?h=su2&id=7dbde4a6a7bddd20e8cfbfdfee2cee4c2ffd8a01) to work with the `meson.py` build script.
